File: Selenium/Activities/Activity22.java
package examples;

import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class Activity22 {

	public static void main(String[] args) {

			WebDriver driver = new FirefoxDriver();
			WebDriverWait explicitWait = new WebDriverWait(driver,Duration.ofSeconds(5));
			
			driver.get("https://training-support.net/webelements/popups");
			System.out.println("The page title is : "+ driver.getTitle());
			
			WebElement btnPopUp =  driver.findElement(By.cssSelector("button#launcher"));
			btnPopUp.click();
			
			explicitWait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("div#popup")));
			
			WebElement txtUsername = driver.findElement(By.cssSelector("[name='username']"));
			WebElement txtPassword = driver.findElement(By.cssSelector("[name='password']"));
			WebElement btnSubmit = driver.findElement(By.xpath("//button[text()='Submit']"));
			
			txtUsername.sendKeys("admin");
			txtPassword.sendKeys("password");
			btnSubmit.click();
			
			explicitWait.until(ExpectedConditions.textToBePresentInElementLocated(By.cssSelector("h2.mt-5"), "Welcome Back, Admin!"));
			
			System.out.println("The message is : "+(driver.findElement(By.cssSelector("h2.mt-5")).getText()));
			
			
			driver.quit();

	}

}

File: Selenium/Activities/Activity22.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
import selenium.webdriver.support.expected_conditions as EC


with webdriver.Firefox() as driver:

    driver.get("https://training-support.net/webelements/popups")
    print(f"The page title is : {driver.title}")

    explicitWait = WebDriverWait(driver,5)

    btnPopUp =  driver.find_element(By.CSS_SELECTOR,"button#launcher")
    btnPopUp.click()

    explicitWait.until(EC.visibility_of_element_located((By.CSS_SELECTOR,"div#popup")))

    txtUsername = driver.find_element(By.CSS_SELECTOR,"[name='username']")
    txtPassword = driver.find_element(By.CSS_SELECTOR,"[name='password']")
    btnSubmit = driver.find_element(By.XPATH,"//button[text()='Submit']")


    txtUsername.send_keys("admin")
    txtPassword.send_keys("password")
    btnSubmit.click()
			
    explicitWait.until(EC.text_to_be_present_in_element((By.CSS_SELECTOR,"h2.mt-5"), "Welcome Back, Admin!"))
    print(f"The message is : {(driver.find_element(By.CSS_SELECTOR,"h2.mt-5")).text}")
    
    driver.quit()


